File: structural/java/adapter/src/RectangleToTriangleAdapter.java
public class RectangleToTriangleAdapter implements Shape{

    private double width;
    private double height;
    private Rectangle rectangle;
    private Triangle triangle;
    public RectangleToTriangleAdapter(Rectangle rectangle){
        this.rectangle = rectangle;
        this.triangle  = new Triangle(this.rectangle.getWidth(), this.rectangle.getHeight());
    }
    @Override
    public void draw() {
        this.triangle.draw();
    }

    @Override
    public double getWidth() {
        return this.width;
    }

    @Override
    public double getHeight() {
        return this.height;
    }
}

File: structural/python/adapter/RectangleToTriangleAdapter.py
from Shape     import Shape
from Rectangle import Rectangle
from Triangle  import Triangle
class RectangleToTriangleAdapter(Shape):
    __rectangle:  Rectangle = None
    __triangle:   Triangle  = None

    def __init__(self, rectangle: Rectangle):
        self.__rectangle = rectangle
        self.__triangle  = Triangle(self.__rectangle.getWidth(), self.__rectangle.getHeight())

    def draw(self) ->None:
        self.__triangle.draw()

    def getWidth(self) ->float:
        return self.__triangle.getWidth()

    def getHeight(self) ->float:
        return self.__triangle.getHeight()